### PR TITLE
fix analyzer errors with 0.24.1

### PR DESCRIPTION
**Pull request by jakemac53**
fixes failures caused by https://code.google.com/p/dart/issues/detail?id=23120

### CODE REVIEW

**Comment by jakemac53** [comment]
tbr @sigmundch

File: lib/src/build/polymer_smoke_generator.dart [hunk 1]
```diff
@@ -223,11 +223,13 @@ class PolymerSmokeGenerator {
   ///    * invoke methods in event handlers.
   _processClass(ClassElement cls, Recorder recorder) {
     if (!_hasPolymerMixin(cls)) return;
+    if (cls.node is! ClassDeclaration) return;
```
Comment (sigmundch) [inline lib/src/build/polymer_smoke_generator.dart]: I think it's safe to remove this line?

**Comment by sigmundch** [comment]
lgtm

File: lib/src/build/polymer_smoke_generator.dart (hunk 1 above)
Comment (jakemac53) [inline lib/src/build/polymer_smoke_generator.dart]: It can now return an EnumDeclaration or ClassDeclaration technically. We need it at least for now.

Comment (sigmundch) [inline lib/src/build/polymer_smoke_generator.dart]: ah, ok, thanks. I just found the bug you filed, I see it will change back in 0.24.2.

**Pull request merged**